types/src/front/lib/api/assistant/generation.ts

```diff
@@ -12,6 +12,53 @@ export type ModelConversationType = {
   messages: ModelMessageType[];
 };
 
+export type ContentFragmentMessageTypeModel = {
```

Comment:
We likely want to get rid of this type entirely and either render that as a fake agent message with function call + response function message or a user message

Reply:
If we decide to have function call + reponse it could look like that: 

```
{
  "role": "assistant",
  "content": null,
  "functionCalls": [
    {
      "id": "81",
      "type": "function",
      "function": {
        "name": "retrieve_content",
        "arguments": "{\"content_type\":\"slack_thread_content\"}" // or file_attachment
      }
    }
  ]
},
{
  "role": "tool",
  "toolCallId": "81",
  "content": "Content from the fragment"
}
```

Or if we decide it's a user message: 

```
{
  "role": "user",
  "name": "get_slack_thread_content", // or get_file_attachment
  "content": "Content from the fragment"
},
```

=> Since some provider are erroring when a convo starts with an agent message cf https://github.com/dust-tt/tasks/issues/699, I will add a commit to set it as a user message. Let me know if you think otherwise!

Reply:
Other providers won’t allow having two user messages in a row I believe 🙃

Reply:
Not sure there’s a great solution here 🤔